### PR TITLE
Docmatix - an instruct dataset for DocVQA

### PR DESCRIPTION
**Pull request by andimarafioti**
First pass at the blog

### CODE REVIEW

File: docmatix.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+authors:
+- user: andito
+- user: HugoLaurencon
+---
+
+# Docmatix - An instruct dataset for Document Visual Question Answering
+
+
+During our work on Idefics2, we developed [The Cauldron](https://huggingface.co/datasets/HuggingFaceM4/the_cauldron), an extensive collection of 50 datasets for the fine-tuning of Vision-Language Model (VLM). Through this process, we identified a significant gap in the availability of large-scale Document Visual Question Answering (DocVQA) datasets. The primary dataset we relied on was DocVQA, which contains 10,000 images and 39,000 question-answer (Q/A) pairs.
+To address this limitation, we are excited to introduce Docmatix, a groundbreaking DocVQA dataset featuring 2.4 million images and 9.5 million Q/A pairs derived from 1.3 million PDF documents. This marks a 240X increase in scale.
```
Comment (HugoLaurencon) [inline docmatix.md:13]: Can you just remove `groundbreaking`? It sounds really chatgpt generated

Comment (HugoLaurencon) [inline docmatix.md:13]:
```suggestion
To address this limitation, we are excited to introduce Docmatix, a DocVQA dataset featuring 2.4 million images and 9.5 million Q/A pairs derived from 1.3 million PDF documents. This marks a 240X increase in scale compared to previous datasets.
```

Comment (andimarafioti) [reply docmatix.md]: Removed it but it actually was my adjective 💔

File: docmatix.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+</div>
+
+<iframe
+  src="https://huggingface.co/datasets/HuggingFaceM4/Docmatix/embed/viewer/default/train"
+  frameborder="0"
+  width="100%"
+  height="560px"
+></iframe>
+
+
```
Comment (HugoLaurencon) [inline docmatix.md:68]: Maybe a last quick sentence for the conclusion?

File: docmatix.md (hunk 1 above)
Comment (HugoLaurencon) [reply docmatix.md]: openai controls you

Comment (andimarafioti) [reply docmatix.md]: I don't delve in such theories

File: docmatix.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+The following plot presents some key statistics from our analysis:
+
+
+<p align="center">
+ <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/docmatix_prompt_analysis.png" alt="Prompt analysis Docmatix" style="width: 90%; height: auto;"><br>
+ <em>Analysis of Docmatix per prompt</em>
+</p>
+
+To evaluate Docmatix's performance, we conducted ablation studies using the Florence-2 model. We trained two versions of the model for comparison. The first version was trained over several epochs on the DocVQA dataset. The second version was trained for one epoch on Docmatix (20% of the images and 4% of the Q/A pairs), followed by one epoch on DocVQA to ensure the model produced the correct format for DocVQA evaluation.
+The results were notable: training on this small portion of Docmatix yielded a relative improvement of almost 20%. Additionally, the 0.7B Florence-2 model performed only 5% worse than the 8B Idefics2 model trained on a mixture of datasets.
```
Comment (lvwerra) [inline docmatix.md:47]:
```suggestion
The results are significant: training on this small portion of Docmatix yielded a relative improvement of almost 20%. Additionally, the 0.7B Florence-2 model performed only 5% worse than the 8B Idefics2 model trained on a mixture of datasets and is significantly larger.
```

File: docmatix.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+# Docmatix - An instruct dataset for Document Visual Question Answering
+
+
+During our work on Idefics2, we developed [The Cauldron](https://huggingface.co/datasets/HuggingFaceM4/the_cauldron), an extensive collection of 50 datasets for the fine-tuning of Vision-Language Model (VLM). Through this process, we identified a significant gap in the availability of large-scale Document Visual Question Answering (DocVQA) datasets. The primary dataset we relied on was DocVQA, which contains 10,000 images and 39,000 question-answer (Q/A) pairs.
+To address this limitation, we are excited to introduce Docmatix, a DocVQA dataset featuring 2.4 million images and 9.5 million Q/A pairs derived from 1.3 million PDF documents. This marks a 240X increase in scale compared to previous datasets.
+
+<div align="center">
+
+| Dataset              | # images | # Q/A pairs | # tokens   |
```
[the remark this answers is not in the document]
Comment (andimarafioti) [reply docmatix.md:17]: Added a beautiful barplot :)

File: docmatix.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+|----------------------|----------|-------------|------------|
+| **Docmatix**         | **2,444,750**| **9,500,000**   | **390,000,000**|
+| DocVQA               | 10,189   | 39,463      | 337,829    |
+| VisualMRC            | 3,027    | 11,988      | 168,828    |
+| InfoVQA              | 2,118    | 10,074      | 61,048     |
+
+</div>
+
+
+Docmatix is generated from PDFA, an extensive OCR dataset containing 2.1 million PDFs. We took the transcriptions from PDFA and employed a Phi-3-small model to generate Q/A pairs. To ensure the dataset's quality, we filtered the generations, discarding 15% of the Q/A pairs identified as hallucinations. To do so, we used regular expressions to detect code and removed answers that contained the keyword “unanswerable”.
```
[the remark this answers is not in the document]
Comment (andimarafioti) [reply docmatix.md:27]: Added the links

File: docmatix.md (hunk 2 above)
Comment (andimarafioti) [reply docmatix.md:84]: Added conclusions

File: docmatix.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+
+
+<div align="center">
+
+| Dataset                              | ANSL on DocVQA |
+|--------------------------------------|----------------|
+| Florence 2 fine-tuned on DocVQA      | 60.1           | 
+| Florence 2 fine-tuned on Docmatix    | 71,4           |
+| Idefics2                             | 74,0           |
```
[the remark this answers is not in the document]
Comment (andimarafioti) [reply docmatix.md]: Added it, good remark!

File: docmatix.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+title: "Docmatix - an instruct dataset for Document Visual Question Answering" 
+thumbnail: /blog/assets/182_finetune-florence/thumbnail.png
+authors:
+- user: andito
+- user: HugoLaurencon
+---
+
+# Docmatix - An instruct dataset for Document Visual Question Answering
+
+
```
[the remark this answers is not in the document]
Comment (andimarafioti) [reply docmatix.md:11]: Added it

**Review by HugoLaurencon — APPROVED** [review]

**Pull request merged**